source/particles/particle_handler.cc

```diff
@@ -1328,15 +1308,6 @@ namespace Particles
 #ifndef DEAL_II_WITH_MPI
     (void)enable_cache;
 #else
-    // First clear the current ghost_particle information
-    for (auto &particles_in_cell : ghost_particles)
-      for (auto &particle : particles_in_cell)
-        if (particle != PropertyPool<dim, spacedim>::invalid_handle)
-          property_pool->deregister_particle(particle);
```

Comment:
Thinking about it some more, where is this happening now in the new framework? Some place still needs to delete the ghost particles that are already on the local process, before communicating and adding the current ones. Otherwise we duplicate ghost particles. Maybe we need the following in this place? You could test if this is necessary by modifying one of the tests to call exchange ghost particles twice. The number of ghost particles should stay the same.
```
for (const auto &ghost_particle = begin_ghost(); ghost_particle != end_ghost(); ++ghost_particle)
  property_pool->deregister_particle(ghost_particle);

for (const auto &cell: triangulation.active_cell_iterators())
  if (cell->is_ghost() == false)
    particles[cell->active_cell_index()].clear();

```

Reply:
You are absolutely right. This is a bit of a nasty bug I was about to introduce since it would have duplicated infinitely the ghost particles. I'll come up with a fix for that.